test/Altinn.App.Core.Tests/Features/Action/SigningUserActionTests.cs

```diff
@@ -31,7 +31,59 @@ public async void HandleAction_returns_ok_if_user_is_valid()
             UserId = 1337,
             Party = new Party() { SSN = "12345678901" }
         };
-        (var userAction, var signClientMock) = CreateSigningUserAction(userProfile);
+        var appMetadata = new ApplicationMetadata("org/id") { DataTypes = [new DataType { Id = "model" }] };
+        (var userAction, var signClientMock) = CreateSigningUserAction(
+            applicationMetadataToReturn: appMetadata,
+            userProfileToReturn: userProfile
+        );
+        var instance = new Instance()
+        {
+            Id = "500000/b194e9f5-02d0-41bc-8461-a0cbac8a6efc",
+            InstanceOwner = new() { PartyId = "5000", },
+            Process = new() { CurrentTask = new() { ElementId = "Task2" } },
+            Data = new()
+            {
+                new() { Id = "a499c3ef-e88a-436b-8650-1c43e5037ada", DataType = "Model" }
+            }
+        };
+        var userActionContext = new UserActionContext(instance, 1337);
+
+        // Act
+        var result = await userAction.HandleAction(userActionContext);
+
+        // Assert
+        SignatureContext expected = new SignatureContext(
+            new InstanceIdentifier(instance),
+            "signature",
+            new Signee() { UserId = "1337", PersonNumber = "12345678901" },
+            new DataElementSignature("a499c3ef-e88a-436b-8650-1c43e5037ada")
+        );
+        signClientMock.Verify(
+            s => s.SignDataElements(It.Is<SignatureContext>(sc => AssertSigningContextAsExpected(sc, expected))),
+            Times.Once
+        );
+        result.Should().BeEquivalentTo(UserActionResult.SuccessResult());
+        signClientMock.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public async void HandleAction_returns_ok_if_no_dataElementSignature_and_optional_datatypes()
+    {
+        // Arrange
+        UserProfile userProfile = new UserProfile()
+        {
+            UserId = 1337,
+            Party = new Party() { SSN = "12345678901" }
+        };
+        var appMetadata = new ApplicationMetadata("org/id")
+        {
+            // Optional because MinCount == 0
+            DataTypes = [new DataType { Id = "model", MinCount = 1 }]
```

Comment:
Is the comment wrong or is the value wrong here?

Reply:
Nice, the comment actually helped uncover a bug. This test was a false positive as the mincount = 1 makes the datatype mandatory, BUT because instance.Data.Datatype includes the Id of the mandatory datatype then its ok.